### PR TITLE
feat: add runtime/sveltekit-support

### PR DESCRIPTION
**Pull request by ryoppippi**
fixes: #3 
fixes: #2

I added runtime import.
This allows this plugin to be used with SvelteKit.

### CODE REVIEW

**Comment by ryoppippi** [comment]
idk how to fix my pr for your preference(like name, pr name or so, it is hard for contributors to follow these rules... I think)
So you can change however you like

**Comment by ryoppippi** [comment]
@prisis please modify as you want

File: examples/sveltekit/src/hooks.server.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import head from "@anolilab/unplugin-favicons/runtime";
+import type { Handle } from "@sveltejs/kit";
+
+export const handle: Handle = async ({ event, resolve }) => {
+    const response = await resolve(event, {
+        transformPageChunk: ({ html }) => html.replace("</head>", `${head}</head>`),
```
Comment (ryoppippi) [inline examples/sveltekit/src/hooks.server.ts:6]: https://kit.svelte.dev/docs/hooks

File: src/core/unplugin-factory.ts [hunk 2]
```diff
@@ -197,14 +208,49 @@ const unpluginFactory: UnpluginFactory<FaviconsIconsPluginOptions | FaviconsLogo
...
+            }
+
+            return {
+                code: s.toString(),
+                map: s.generateMap({
+                    includeContent: true,
+                    source: id,
+                }),
+            };
+        },
```
Comment (ryoppippi) [inline src/core/unplugin-factory.ts:242]: Based on this code: https://github.com/unjs/magic-regexp/blob/main/src/transform.ts

**Comment by ryoppippi** [comment]
@prisis please modify as you want

File: examples/sveltekit/svelte.config.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import adapter from "@sveltejs/adapter-auto";
-import { vitePreprocess } from "@sveltejs/kit/vite";
+import { vitePreprocess } from "@sveltejs/vite-plugin-svelte";
```
Comment (ryoppippi) [inline examples/sveltekit/svelte.config.js:2]: https://kit.svelte.dev/docs/migrating-to-sveltekit-2#vitepreprocess-is-no-longer-exported-from-sveltejs-kit-vite

**Comment by ryoppippi** [comment]
The problem is that the favicon doesn't show up on dev mode

**Comment by prisis** [comment]
I would not close it, its still a good feature

**Comment by ryoppippi** [comment]
Wait i cannot reopen this

**Comment by prisis** [comment]
Because you deleted your branch 😅 will try the next days to restore it :)

**Pull request closed without merging**